### ISSUE TITLE
upgrade-with-expansion option clashes with the expansions option

### ISSUE THREAD

**Issue by md5i** [sumpfork/dominiontabs#383]
If you include `--upgrade-with-expansion`, all expansions with upgraded cards are output even if those expansions aren't included in a given `--expansions` option.  So, for example, if you do:
```
dominion_dividers --upgrade-with-expansion --expansions "intrigue"
```
It will include all the Dominion 1st and 2nd edition cards along with the Intrigue ones.

**Comment by md5i** [comment]
I looked at creating a PR for this, but `--upgrade-with-expansion` is handled very early in `filter_sort_cards`, and `--expansions` is handled very late.  This part of the code is complicated enough that I don't know what I might break by moving things around.

**Comment by wvoigt** [comment]
I lost my development/test system so I cannot verify this.  But I think the issue is here:
```
    # Combine upgrade cards with their expansion
    if options.upgrade_with_expansion:
        for card in cards:
            if Card.sets[card.cardset_tag]["upgrades"]:
                card.cardset_tag = Card.sets[card.cardset_tag]["upgrades"]
                options.expansions.append(card.cardset_tag.lower())         <----- problem here
```
If it is a card from an upgrade, then it first assigns the card to the expansion set it is upgrading.  Which is what you would expect for --upgrade-with-expansion.
But the last line then adds the expansion on to the list of expansions to print.  This should not be needed and is creating the wrong behavior.  This line should be removed.

**Comment by md5i** [comment]
I traced it to this line as well but am hesitant to say more, as I doubt this was added for no reason whatsoever.  If I were to guess, it's so that if, say, Intrigue was selected, that with the `--upgrade-with-expansion` option would automatically include the "Intrigue 2nd Edition" expansion.  The implementation here is obviously flawed, but I don't know that *just* removing that line is correct.

**Comment by wvoigt** [comment]
Why would you want "Intrigue 2nd Edition" to print when you upgraded the 1st edition?  This was clearly done under the false assumption that if this option was selected the person would want all the updated sets printed as well. As we are going to have more of these, I think that logic is false. We should let this option do as it says, and no more. Let the determination of what sets to print be handled by the other options.

**Comment by md5i** [comment]
After thinking about it, I think you are right.

**Comment by wvoigt** [comment]
With some more thought, one thing that is missing if `--upgrade-with-expansion` is used is the **removal** of the upgrade set from the list of sets/expansions to print.  For example, if you are printing everything, and one has `--upgrade-with-expansion` you would not expect those cards to show up again in their own set/expansion since they were asked to be included elsewhere.
So it is possible, way back when this was added that this was simply coded wrong.  I think it would look like:
```
    # Combine upgrade cards with their expansion
    if options.upgrade_with_expansion:
        for card in cards:
            if Card.sets[card.cardset_tag]["upgrades"]:
                options.exclude_expansions.append(card.cardset_tag.lower())  
                card.cardset_tag = Card.sets[card.cardset_tag]["upgrades"]

``` 
Again, my development system is unavailable, so I am not able to test this.

**Comment by wvoigt** [comment]
This was fixed with Pull #391 along with the addition of the removed cards from the 1st edition expansions, starting with Pull #393 .  So closing.  If something was missed, then it can be reopened.
